tapdb/assets_common.go

```diff
@@ -200,6 +204,24 @@ func upsertAssetsWithGenesis(ctx context.Context, q UpsertAssetStore,
 			anchorUtxoID = anchorUtxoIDs[idx]
 		}
 
+		// Check for matching assets in the database. If we find one,
+		// we'll just use its primary key ID, and we won't attempt to
+		// insert the asset again.
+		existingAssets, err := q.QueryAssets(ctx, QueryAssetFilters{
+			AnchorUtxoID: anchorUtxoID,
+			GenesisID:    sqlInt64(genAssetID),
+			ScriptKeyID:  sqlInt64(scriptKeyID),
+		})
+		if err != nil {
+			return 0, nil, fmt.Errorf("unable to query assets: %w",
+				err)
+		}
+
+		if len(existingAssets) > 0 {
```

Comment:
IIUC the returned asset primary keys here are not in a stable order, e.x. if I already have duplicate assets, we don't have a guarantee about the same asset ID always being returned here.

However, I don't think this is problem, as the only place I see the `assetIDs` array returned by `upsertAssetsWithGenesis` being used is here:

https://github.com/lightninglabs/taproot-assets/blob/b8a007d3cb14de85b03dabb606c248491a57b491/tapdb/assets_store.go#L1445

Where we're inserted one asset, which should be new. TL;DR the unstable ordering should be fine.